packages/vm/src/middleware/deterministic.rs

```diff
@@ -138,14 +138,386 @@ impl FunctionMiddleware for FunctionDeterministic {
             | Operator::I64Extend8S
             | Operator::I64Extend16S
             | Operator::I64ExtendI32S
-            | Operator::I64ExtendI32U => {
+            | Operator::I64Extend32S
+            | Operator::I64ExtendI32U
+            | Operator::RefNull { .. }
+            | Operator::RefIsNull
+            | Operator::RefFunc { .. }
+            | Operator::ReturnCall { .. }
+            | Operator::ReturnCallIndirect { .. }
+            | Operator::TypedSelect { .. }
+            | Operator::TableGet { .. }
+            | Operator::TableSet { .. }
+            | Operator::TableGrow { .. }
+            | Operator::TableSize { .. } => {
```

Comment:
Those seem to be a proposed extension as well: https://github.com/WebAssembly/reference-types/blob/master/proposals/reference-types/Overview.md

Reply:
So separate error for these as well? Done.